.devcontainer/devcontainer.json

```
@@ -0,0 +1,47 @@
+// For format details, see https://aka.ms/devcontainer.json. For config options, see the
+// README at: https://github.com/devcontainers/templates/tree/main/src/ubuntu
+{
+	"name": "Ubuntu",
+	// Or use a Dockerfile or Docker Compose file. More info: https://containers.dev/guide/dockerfile
+	"image": "mcr.microsoft.com/devcontainers/base:jammy",
+	"features": {
+		"ghcr.io/devcontainers/features/node:1": {
+			"nodeGypDependencies": true,
+			"version": "lts",
+			"nvmVersion": "latest"
+		},
+		"ghcr.io/devcontainers/features/python:1": {
+			"installTools": true,
+			"version": "3.12.2",
```

```suggestion
			"version": "3.12",
```

Same here, as you are not using `universal`, feel free to pin it to `3.12` (you no longer need to pin to a specific patch version)

---

Should we be using 3.11 for the container? That is the minimum version for typechat.

---

It's the minimum, but I will work on making the workspace 3.12-friendly so that we can make sure 3.12 tests run correctly.